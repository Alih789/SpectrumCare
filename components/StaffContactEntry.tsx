import {Image, StyleSheet, Text, View, Pressable, Linking} from 'react-native'
import Ionicons from 'react-native-vector-icons/Ionicons';
import React, {useState} from 'react'

type itemProps ={name: string, imagePath: string, jobTitle: string, department: string, onPress: (isPressed: boolean) => void, hyperlink: string}

function StaffContactEntry({name, imagePath, jobTitle, department, onPress, hyperlink}: itemProps): JSX.Element {

  const [isPressed, setIsPressed] = useState(false);
  
  const handleFavPress = () => {
    setIsPressed(!isPressed);
    if (onPress) {
      onPress(!isPressed);
    }
  }

  return (
    <View style={styles.container}>
      <Image source={{uri: imagePath}} style={styles.image}/>
      <View style={styles.detailsContainer}>
        <Pressable style={styles.favPostioning} onPress={handleFavPress}>
          <Ionicons name="heart" style={[styles.defaultFav, isPressed && styles.favButtonPressed]} size={22}/>
        </Pressable>
        <Text style={styles.name}>{name}</Text>
        <Text style={styles.jobTitle}>{jobTitle}</Text>
        <Text style={styles.department}>{department}</Text>
        <Pressable onPress={() => Linking.openURL(hyperlink)}>
          <View style={styles.box}>
            <Text style={styles.linkText}>View Full Profile {">"} </Text>
          </View>
        </Pressable>
      </View>
    </View>
  )
}

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    alignItems: 'center',
    padding: 10,
    height: 200,
    backgroundColor: 'white',
    borderRadius: 15,
    borderWidth: 2,
  },
  image: {
    width: 100,
    height: 150,
    borderRadius: 10,
    marginRight: 10,
  },
  detailsContainer: {
    flexDirection: 'column',
    flexShrink: 1,
    // position: 'relative',
  },
  favPostioning:{
    top: -25,
    right: 0,
    position: "absolute",

  },
  defaultFav: {
    color: 'grey',
  },
  favButtonPressed: {
    color: 'red',
  },
  name: {
    flexWrap: "wrap",
    fontWeight: 'bold',
    fontSize: 14,
    paddingBottom: 3,
  },
  jobTitle: {
    flexWrap: "wrap",
    fontWeight: "600",
    fontSize: 10,
  },
  department: {
    flex: 0.5,
    flexWrap: "wrap",
    fontSize: 9,
  },
  box: {
    borderWidth: 2,
    borderColor: '#003A5D',
    padding: 10,
    borderRadius: 8,
  },
  linkText: {
    textAlign: 'center',
    fontWeight: "bold",
    color: "#003A5D",
  },
})
export default StaffContactEntry