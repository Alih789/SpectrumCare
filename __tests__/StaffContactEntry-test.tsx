/** @jest-environment jsdom */
import 'react-native';
import React from 'react';
import StaffContactEntry from '../components/StaffContactEntry';
import renderer from 'react-test-renderer';


describe("StaffContactEntry", () => {

    //Button was pressed
    const handlePress = () => {
        return true;
    }
    //isFavorited set to true to match the color
    test("renders StaffContactEntry", () => {
        const tree = renderer.create(
            <StaffContactEntry
            name='Angel Alberto Herrera Guerra, M.D.'
            imagePath='https://physicians.ucdavis.edu/Custom/Photos/32290.jpg'
            jobTitle='Associate Professor'
            department='Pediatric Allergy, Immunology and Rheumatology'
            phoneNumber='916-734-3112'
            onPress={handlePress}
            isFavorited={true}
            hyperlink='https://health.ucdavis.edu/pediatrics/team/32290/angel_alberto-herrera_guerra-pediatric_infectious_diseases'
            />
            ).toJSON();
        expect(tree).toMatchSnapshot();
    });
  });
