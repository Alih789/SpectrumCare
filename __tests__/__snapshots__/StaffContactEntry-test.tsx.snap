// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StaffContactEntry renders StaffContactEntry 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "white",
      "borderColor": "white",
      "borderRadius": 15,
      "borderWidth": 2,
      "elevation": 3,
      "flexDirection": "row",
      "flexShrink": 1,
      "height": 200,
      "justifyContent": "space-between",
      "marginBottom": 10,
      "padding": 10,
      "shadowColor": "black",
      "shadowOffset": {
        "height": 1,
        "width": 0,
      },
      "shadowOpacity": 4,
      "shadowRadius": 2.22,
    }
  }
>
  <Image
    source={
      {
        "uri": "https://physicians.ucdavis.edu/Custom/Photos/32290.jpg",
      }
    }
    style={
      {
        "borderRadius": 10,
        "height": 150,
        "marginRight": 10,
        "width": 100,
      }
    }
  />
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "position": "absolute",
        "right": 5,
        "top": 5,
      }
    }
  >
    <Text
      allowFontScaling={false}
      selectable={false}
      style={
        [
          {
            "color": undefined,
            "fontSize": 22,
          },
          [
            {
              "color": "red",
            },
          ],
          {
            "fontFamily": "Ionicons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          {},
        ]
      }
    >
      ïŒ®
    </Text>
  </View>
  <View
    style={
      {
        "flexDirection": "column",
        "marginLeft": 10,
        "marginRight": 10,
        "maxHeight": "100%",
        "maxWidth": 200,
        "paddingRight": 10,
      }
    }
  >
    <Text
      selectable={true}
      style={
        {
          "flexWrap": "wrap",
          "fontFamily": "Figtree-Medium",
          "fontSize": 17,
          "fontWeight": "bold",
          "lineHeight": 17,
          "marginBottom": 3,
        }
      }
    >
      Angel Alberto Herrera Guerra, M.D.
    </Text>
    <Text
      selectable={true}
      style={
        {
          "flexWrap": "wrap",
          "fontFamily": "Figtree-Medium",
          "fontSize": 12,
          "fontWeight": "600",
          "lineHeight": 17,
          "marginBottom": 5,
        }
      }
    >
      Associate Professor
    </Text>
    <Text
      selectable={true}
      style={
        {
          "flexWrap": "wrap",
          "fontFamily": "Figtree-Medium",
          "fontSize": 13,
          "lineHeight": 17,
          "marginBottom": 5,
        }
      }
    >
      Pediatric Allergy, Immunology and Rheumatology
    </Text>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
    >
      <Text
        selectable={true}
        style={
          {
            "color": "black",
            "fontFamily": "Figtree-Light",
            "fontSize": 13,
            "fontWeight": "500",
            "marginBottom": 12,
          }
        }
      >
        Office: 
        916-734-3112
      </Text>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
    >
      <View
        style={
          {
            "borderColor": "#003A5D",
            "borderRadius": 8,
            "borderWidth": 2,
            "marginRight": 10,
            "paddingBottom": 5,
            "paddingLeft": 20,
            "paddingRight": 20,
            "paddingTop": 5,
          }
        }
      >
        <Text
          style={
            {
              "color": "#003A5D",
              "fontFamily": "Figtree-Light",
              "fontWeight": "bold",
              "textAlign": "center",
            }
          }
        >
          View Full Profile 
          &gt;
           
        </Text>
      </View>
    </View>
  </View>
</View>
`;
